api/types/desired_state_test.go

```diff
@@ -187,3 +187,188 @@ func TestFromDesiredStateBytes(t *testing.T) {
 	assert.Equal(t, "os-image", selfupdate.Components[0].ID)
 	assert.Equal(t, "1.10-alpha", selfupdate.Components[0].Version)
 }
+
+func TestSplitPerDomains(t *testing.T) {
+	t.Run("test_split_per_domains", func(t *testing.T) {
+		desiredState := &DesiredState{
```

Comment:
this `desiredState` looks the same as `test_split_per_domains_nil_domains`, please reuse it